<?xml version="1.0"?>

<launch>
  <env name="GAZEBO_RESOURCE_PATH" value="$(find amr_gazebo)/media/materials"/>

  <arg name="use_rviz" default="false" /> 
  <arg name="use_joy" default="false" /> 
  <arg name="gazebo_gui" default="true" /> 

<!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
<?ignore
    <arg name="world_name" value="$(find amr_gazebo)/worlds/factory_sim.world"/>
?>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg gazebo_gui)"/>
    <arg name="recording" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="verbose" value="false"/>
  </include>


<!-- Spawn solamr_1 in Gazebo -->
<arg name="robot_name" value="solamr_1"/>
<param name="tf_prefix" value="$(arg robot_name)" />  
<!-- interpret xacro into urdf-->
<param name="robot_description" command="$(find xacro)/xacro --inorder $(find amr_description)/urdf/solamr_rod.urdf.xacro robotName:=$(arg robot_name)" />
<!-- spawn model from parameter server : robot_description -->
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf_$(arg robot_name)" args="-urdf -model $(arg robot_name) -x -2 -y 1 -z 0 -param robot_description" />
<!-- control file: controller spawner, robot_state_publisher -->
<include file="$(find amr_control)/launch/only_one_control.launch">
    <arg name="robot_ns" value="$(arg robot_name)" />
</include>


<!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find amr_gazebo)/rviz/single_solamr.rviz" if="$(arg use_rviz)" />


<!-- joy teleop -->
    <!--
    <arg name="single" default="true"/>
    -->

    <arg name="joy_maxVel" value=".5" />
    <arg name="joy_accele" value=".01"/>

    <!-- solamr_1 -->
    <node pkg="amr_gazebo" type="teleop_joy.py" name="solamr_1_teleop" output="screen" if="$(arg use_joy)">
        <param name="robot_ns" value="solamr_1" />
        <param name="max_velocity" value="$(arg joy_maxVel)" />
        <param name="acceleration" value="$(arg joy_accele)" />
    </node>

    <node pkg="joy" type="joy_node" name="solamr_1_joy_node" if="$(arg use_joy)">
        <param name="dev" type="string" value="/dev/input/js0" />
        <remap from="/joy" to="/solamr_1/joy" />
	    <param name="deadzone" value="0.0" />
	    <param name="autorepeat_rate" value="130" />
    </node>

<!-- keyboard teleop -->
    <node pkg="amr_gazebo" type="teleop_key.py" name="solamr_teleop" output="screen" unless="$(arg use_joy)">
    </node>

</launch>
